Library/ViewModels/DiscoveryPostcardViewModel.swift

```diff
@@ -133,10 +150,20 @@ public protocol DiscoveryPostcardViewModelType {
 public final class DiscoveryPostcardViewModel: DiscoveryPostcardViewModelType,
   DiscoveryPostcardViewModelInputs, DiscoveryPostcardViewModelOutputs {
 
-    public init() {
-    let project = self.projectProperty.signal.skipNil()
+  // swiftlint:disable function_body_length
+  public init() {
+    let configuredProject = self.projectProperty.signal.skipNil()
+      .map(cached(project:))
+
+    let currentUser = Signal.merge([
+      self.userSessionStartedProperty.signal,
+      self.userSessionEndedProperty.signal,
+      configuredProject.ignoreValues()
```

Comment:
When we do not have a current user when the project emits we prompt the user to login when they tap on the heart/save button. Otherwise, if we have a user once projects emit we do not have to show the login prompt when they tap the heart/save button. @theginbin